prov/shm/src/smr_signal.h

```diff
@@ -62,8 +62,9 @@ static void smr_handle_signal(int signum, siginfo_t *info, void *ucontext)
 	/* call the original handler */
 	if (old_action[signum].sa_flags & SA_SIGINFO)
 		old_action[signum].sa_sigaction(signum, info, ucontext);
-	else if (old_action[signum].sa_handler == SIG_DFL ||
-		 old_action[signum].sa_handler == SIG_IGN)
+	else if (old_action[signum].sa_handler == SIG_DFL)
+		raise(signum);
+	else if (old_action[signum].sa_handler == SIG_IGN)
 		return;
 	else
 		old_action[signum].sa_handler(signum);
```

Comment:
Would it be simpler to replace L65~L70 with:
```
	else
		raise(signum);
```

Reply:
@shefty I agree with Jianxin. This seems simpler to me. But yes, we should include in 1.15 and probably cherry pick to 1.14 as well.

Reply:
@shefty I agree with Jianxin. This seems simpler to me. But yes, we should include in 1.15 and probably cherry pick to 1.14 as well.